packages/application-generic/src/services/readiness.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { BullMqService } from './bull-mq.service';
+
+import {
+  QueueServiceHealthIndicator,
+  TriggerQueueServiceHealthIndicator,
+  WsQueueServiceHealthIndicator,
+} from '../health';
+
+export interface INovuWorker {
+  name: string;
+  bullMqService: BullMqService;
+  gracefulShutdown: () => Promise<void>;
+  pauseWorker: () => Promise<void>;
+  resumeWorker: () => Promise<void>;
+}
+
+const LOG_CONTEXT = 'ReadinessService';
+
+@Injectable()
+export class ReadinessService {
+  constructor(
+    private queueServiceHealthIndicator: QueueServiceHealthIndicator,
+    private triggerQueueServiceHealthIndicator: TriggerQueueServiceHealthIndicator,
+    private wsQueueServiceHealthIndicator: WsQueueServiceHealthIndicator
+  ) {}
+
+  async areQueuesEnabled(): Promise<boolean> {
+    Logger.log('Enabling queues as workers are meant to be ready', LOG_CONTEXT);
+
+    try {
+      const healths = await Promise.all([
+        this.queueServiceHealthIndicator.isHealthy(),
+        this.triggerQueueServiceHealthIndicator.isHealthy(),
+        this.wsQueueServiceHealthIndicator.isHealthy(),
+      ]);
+
+      return healths.every((health) => !!health === true);
```

Comment:
In a µservices world I would do HTTP/gRPC calls to the health check endpoints of the different services we are dependant. In this case we share the health indicators in our app generic package. Underneath all them use the same BullMQ instance with its dependency to a dedicated in-memory database. So we check the 3 even though with one would be enough. This would let it prepared if we create a dedicate in-memory instance for every queue in the future.